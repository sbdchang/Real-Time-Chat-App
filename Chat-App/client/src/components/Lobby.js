import React from 'react';

const Lobby = ({
  username,
  handleUsernameChange,
  roomName,
  handleRoomNameChange,
  handleSubmit
}) => {
  const queryString = window.location.search;
  console.log(queryString);
  const urlParams = new URLSearchParams(queryString);
  const room_name = urlParams.get('room_name');
  const user_name = urlParams.get('user_name');
  console.log(user_name);
  handleUsernameChange(user_name);
  handleRoomNameChange(room_name);
  handleSubmit();
  return (
    <form onSubmit={handleSubmit}>
      <h2>Enter a room</h2>
      <div>
        <label htmlFor="name">Name:</label>
        <input
          type="text"
          id="field"
          value={username}
          onChange={handleUsernameChange}
          required
        />
      </div>

      <div>
        <label htmlFor="room">Room name:</label>
        <input
          type="text"
          id="room"
          value={roomName}
          onChange={handleRoomNameChange}
          required
        />
      </div>

      <button type="submit">Submit</button>
    </form>
  );
};

export default Lobby;
