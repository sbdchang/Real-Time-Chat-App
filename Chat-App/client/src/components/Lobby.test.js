import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import Lobby from './Lobby';
import renderer from 'react-test-renderer';


it('Lobby snapshot', () => {
  const tree = renderer.create(<Lobby />).toJSON();
  expect(tree).toMatchSnapshot();
});

