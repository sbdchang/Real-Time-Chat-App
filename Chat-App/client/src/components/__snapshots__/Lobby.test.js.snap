// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby snapshot 1`] = `
<form>
  <h2>
    Enter a room
  </h2>
  <div>
    <label
      htmlFor="name"
    >
      Name:
    </label>
    <input
      id="field"
      required={true}
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="room"
    >
      Room name:
    </label>
    <input
      id="room"
      required={true}
      type="text"
    />
  </div>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
